dotnet/src/Connectors/Connectors.Memory.Kusto/KustoMemoryStore.cs

```diff
@@ -0,0 +1,411 @@
+        this._queryClient = KustoClientFactory.CreateCslQueryProvider(kcsb);
+        this._adminClient = KustoClientFactory.CreateCslAdminProvider(kcsb);
+    }
+
+    /// <inheritdoc/>
+    public async Task CreateCollectionAsync(string collectionName, CancellationToken cancellationToken = default)
+    {
+        await this.InitializeVectorFunctionsAsync().ConfigureAwait(false);
+
+        using var resp = await this._adminClient
+            .ExecuteControlCommandAsync(
+                this._database,
+                CslCommandGenerator.GenerateTableCreateCommand(new TableSchema(GetTableName(collectionName), s_collectionColumns)),
+                GetClientRequestProperties()
+            ).ConfigureAwait(false);
+    }
+
+    /// <inheritdoc/>
+    public async Task DeleteCollectionAsync(string collectionName, CancellationToken cancellationToken = default)
+    {
+        using var resp = await this._adminClient
+            .ExecuteControlCommandAsync(
+                this._database,
+                CslCommandGenerator.GenerateTableDropCommand(GetTableName(collectionName)),
+                GetClientRequestProperties()
+            ).ConfigureAwait(false);
+    }
+
+    /// <inheritdoc/>
+    public async Task<bool> DoesCollectionExistAsync(string collectionName, CancellationToken cancellationToken = default)
+    {
+        var result = await this._adminClient
+            .ExecuteControlCommandAsync<TablesShowCommandResult>(
+                this._database,
+                CslCommandGenerator.GenerateTablesShowCommand() + $" | where TableName == '{GetTableName(collectionName)}'",
+                GetClientRequestProperties()
+            ).ConfigureAwait(false);
+
+        return result.Count() == 1;
+    }
+
+    /// <inheritdoc/>
+    public async Task<MemoryRecord?> GetAsync(string collectionName, string key, bool withEmbedding = false, CancellationToken cancellationToken = default)
+    {
+        var result = this.GetBatchAsync(collectionName, new[] { key }, withEmbedding, cancellationToken);
+        return await result.FirstOrDefaultAsync(cancellationToken).ConfigureAwait(false);
+    }
+
+    /// <inheritdoc/>
+    public async IAsyncEnumerable<MemoryRecord> GetBatchAsync(string collectionName, IEnumerable<string> keys, bool withEmbeddings = false, [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        var inClauseValue = string.Join(",", keys.Select(k => $"'{k}'"));
+        var query = $"{this.GetBaseQuery(collectionName)} " +
+            $"| where Key in ({inClauseValue}) " +
+            "| project " +
+            // Key
+            $"{s_collectionColumns[0].Name}, " +
+            // Metadata
+            $"{s_collectionColumns[1].Name}=tostring({s_collectionColumns[1].Name}), " +
+            // Timestamp
+            $"{s_collectionColumns[3].Name}, " +
+            // Embedding
+            $"{s_collectionColumns[2].Name}=tostring({s_collectionColumns[2].Name})";
```

Comment:
Accessing column names by using array with index numbers is not very readable, so you had to add comments. 

We can initialize each column separately, so in this case we will be able to access specific column by name and avoid index numbers and comments.

Definition:
```csharp
private static readonly ColumnSchema KeyColumn = new("Key", typeof(string).FullName);
private static readonly ColumnSchema MetadataColumn = new("Metadata", typeof(object).FullName);
private static readonly ColumnSchema EmbeddingColumn = new("Embedding", typeof(object).FullName);
private static readonly ColumnSchema TimestampColumn = new("Timestamp", typeof(DateTime).FullName);

private static readonly ColumnSchema[] s_collectionColumns = new ColumnSchema[]
{
    KeyColumn,
    MetadataColumn,
    EmbeddingColumn,
    TimestampColumn
};
```

Usage:
```csharp
var query = $"{this.GetBaseQuery(collectionName)} " +
    $"| where Key in ({inClauseValue}) | project " +
    $"{KeyColumn.Name}, " +
    $"{MetadataColumn.Name}=tostring({MetadataColumn.Name}), " +
    $"{TimestampColumn.Name}, " +
    $"{EmbeddingColumn.Name}=tostring({EmbeddingColumn.Name})";
```